model/service.go

```diff
@@ -57,6 +57,33 @@ type Service struct {
 // Tag is a non-empty set of key-value pairs.
 type Tag map[string]string
 
+// SubsetOf is true if the tag has identical values for the keys
+func (tag Tag) SubsetOf(that Tag) bool {
+	for k, v := range tag {
+		if that[k] != v {
+			return false
+		}
+	}
+	return true
+}
+
+// TagList is a set of tags
+type TagList []Tag
+
+// Matches returns true if the input tag is a super set of one of the tags in the list
+// or if the tag list if empty
+func (tags TagList) Matches(that Tag) bool {
+	if len(tags) == 0 {
```

Comment:
This is an odd semantics for a matches function (returning true when one side is empty). While it may simplify code, it might make it hard to understand unless one remembers that empty can match with a non empty set. 

I suggest renaming. Like subsetOf here as well.